src/Middleware.ts

```diff
@@ -100,18 +100,52 @@ export function sessionMiddleware(options: SessionOptions) {
 
     await next()
 
-    if (c.get('session_key_rotation') === true && !(store instanceof CookieStore)) {
-      await store.deleteSession(sid)
-      sid = await nanoid(21)
-      await store.createSession(sid, session.getCache())
-
-      setCookie(c, sessionCookieName, encryptionKey ? await encrypt(encryptionKey, sid) : sid, cookieOptions)
+    const shouldDelete = session.getCache()._delete;
+    const shouldRotateSessionKey = c.get("session_key_rotation") === true;
+    const storeIsCookieStore = store instanceof CookieStore;
```

Comment:
The original conditionals were logical but a bit challenging for me to grok. I'm a big fan of abstracting conditionals into well named variables. I personally lean towards verbosity over succinctness to make the intention really clear.

That being said, happy to make changes if there is different stylistic preference on how to write these conditionals.